uktrade/data-hub-frontend
# Reactify OMIS payment reconciliation page

## Description of change

The form for reconciling an OMIS payment has been reactified.

## Test instructions

With your local FE pointing to dev, go to the [OMIS reconciliation list](http://localhost:3000/omis/reconciliation?sortby=payment_due_date%3Aasc&status=quote_accepted) and click on an order. You should be able to reconcile the payment. Note that some of the orders within dev don't have invoices, which will raise an error. There aren't any orders in production at the `quote_accepted` stage without invoices, so this can be disregarded.

View an order with the status set to `draft` or `quote_awaiting_acceptance` (you can find these by unticking all the status filters on the reconciliation list). You should see a message stating that the payment cannot be reconciled.

View an order with the status set to `paid` or `complete`. You should see a message stating that the order has already been paid for and a link to the receipt.

View an order with the status set to `cancelled`. You should see a message stating that the payment cannot be reconciled because the order has been cancelled.

## Screenshots

### Before

![Screenshot 2023-10-02 at 12-38-33 Department for Business and Trade](https://github.com/uktrade/data-hub-frontend/assets/36161814/fa73584f-2d06-4b16-a08e-5062000cc688)

### After

![Screenshot 2023-10-02 at 12-41-08 Reconcile a payment - CTU296_22 - Orders (OMIS) - DBT Data Hub](https://github.com/uktrade/data-hub-frontend/assets/36161814/3d2d61b8-2bac-4783-98c7-327ac62b029a)


## Checklist

[//]: # 'When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/main/docs/Code%20review%20guidelines.md'
